langchain4j-mcp/src/test/java/dev/langchain4j/mcp/client/transport/http/McpHttpTransportTest.java

```diff
@@ -0,0 +1,123 @@
+package dev.langchain4j.mcp.client.transport.http;
+
+import static dev.langchain4j.model.openai.OpenAiChatModelName.GPT_4_O_MINI;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.mcp.McpToolProvider;
+import dev.langchain4j.mcp.client.DefaultMcpClient;
+import dev.langchain4j.mcp.client.McpClient;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.tool.ToolExecutor;
+import dev.langchain4j.service.tool.ToolProvider;
+import dev.langchain4j.service.tool.ToolProviderResult;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+@EnabledIfEnvironmentVariable(named = "OPENAI_API_KEY", matches = ".+")
```

Comment:
Why McpHttpTransport test needs AI model? What is the system under test? Integration test should have IT suffix.

The project already has way too many integration tests that are not being executed for PRs, because they are slow and expensive. 

Would you mind to keep this test only focusing on McpHttpTransport testing? LLM integration is tested somewhere else, as long as McpToolProivder produce tools complying with the ToolSpecification

Reply:
I definitely could replace the openai model with a mock, it will just be a bunch of work, I thought it easier to just use the real model... Wdyt @langchain4j ?

Reply:
Hm, `ChatModelMock` only serves a single hardcoded response, and I need it to request tools etc.

Reply:
Yeah I agree, no need to use real model here, you could either mock `ChatLanguageModel` interface or add into `ChatModelMock` a `thatAlwaysResponds(AiMessage)` method that will allow you to specify tools

Reply:
I have an idea and I will rewrite it so that it doesn't need a chat model at all.

Reply:
Done, please check this test again